riscv/csrs.cc

```diff
@@ -1561,6 +1555,17 @@ void senvcfg_csr_t::verify_permissions(insn_t insn, bool write) const {
   masked_csr_t::verify_permissions(insn, write);
 }
 
+// senvcfg.sse is read_only 0 when menvcfg.sse = 0
+reg_t senvcfg_csr_t::read() const noexcept {
+  return (((state->v)? state->henvcfg->read() : state->menvcfg->read()) |
+      (~MENVCFG_SSE)) & masked_csr_t::read();
```

Comment:
Should this have been:
return (masked_csr_t::read() & ((state->v ? state->henvcfg->read() : state->menvcfg->read()) & MENVCFG_SSE ? -1 : ~MENVCFG_SSE));

Reply:
I am not sure why its better to use the scope operator instead of derefencing the object.  Its also not clear to me why it is better to use scope operator for invoking read method of henvcfg class but dereferencing the menvcfg object is okay. I am not that versed with C++ nuances.

Reply:
I agree your point, and I will revise it, thanks